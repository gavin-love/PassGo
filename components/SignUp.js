import React, { Component } from 'react';
import { View, TextInput, Text, Button, StyleSheet } from 'react-native';
import { auth } from '../firebase';

class SignUp extends Component {
  constructor() {
    super();
    this.unsubscriber = null;
    this.state = {
      isAuthenticated: false,
      email: '',
      password: '',
      userName: ''
    };
  }

  postNewUser = (user) => {
    return fetch('https://pass-go.herokuapp.com/api/v1/users', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ user: { email: user.user.email, name: user.user.displayName } })
    })
      .then(response => response.json())
      .then(result => console.log(result))
      .catch(error => console.log(error.message));
  }

  handleSignUp = () => {
    return auth
      .doCreateUserWithEmailAndPassword(
        this.state.email,
        this.state.password
      )
      .then(user => {
        if (user) {
          return user.user.updateProfile({
            displayName: this.state.userName
          }).then(() => this.postNewUser(user))
        }
      })
      .then(() => {
        this.setState({
          email: '',
          password: '',
          userName: ''
        })
      })
      .catch(error => {
        console.log(error.message)
      });
  }

  render() {
    return (
      <View style={styles.page}>
        <View>
          <Text style={styles.title}>Enter account info here:</Text>
          <Text style={styles.name}>User Name</Text>
          <TextInput name='user-name' value={this.state.userName} onChangeText={(text) => this.setState({ userName: text })} style={styles.textInput} />
          <Text style={styles.email}>Email</Text>
          <TextInput name='email' value={this.state.email} onChangeText={(text) => this.setState({ email: text })} style={styles.textInput} />
          <Text style={styles.password}>Password</Text>
          <TextInput name='password' value={this.state.password} onChangeText={(text) => this.setState({ password: text })} style={styles.textInput} />
          <View style={styles.submit}>
            <Button title="sign up" color="#f3f3f3" onPress={() => {
              this.handleSignUp()
                .then(() => this.props.navigation.navigate('Home'));
            }} />
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  page: {
    alignItems: 'center',
    backgroundColor: '#3383bb',
    height: '100%'
  },
  title: {
    fontSize: 30,
    width: 200,
    color: '#f3f3f3'
  },
  name: {
    color: '#f3f3f3',
    marginTop: 10,
    marginBottom: 5
  },
  email: {
    color: '#f3f3f3',
    marginTop: 10,
    marginBottom: 5
  },
  password: {
    color: '#f3f3f3',
    marginTop: 10,
    marginBottom: 5
  },
  textInput: {
    height: 40,
    borderColor: 'grey',
    borderWidth: 1,
    width: 200
  },
  submit: {
    justifyContent: 'center',
    borderColor: 'grey',
    borderWidth: 1,
    height: 40,
    width: 200,
    borderRadius: 5,
    marginTop: 10,
    shadowColor: '#f3f3f3',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  }
});

export default SignUp;