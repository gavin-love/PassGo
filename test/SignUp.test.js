import React from 'react';
import SignUp from '../components/SignUp';
import TestRenderer from 'react-test-renderer';

it('should match snapshot upon render', () => {
  const navigation = {
    navigate: jest.fn()
  }
  const wrapper = TestRenderer.create(<SignUp navigation={navigation}/>);
  expect(wrapper).toMatchSnapshot();
})