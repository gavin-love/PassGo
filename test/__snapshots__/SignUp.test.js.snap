// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot upon render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3383bb",
      "height": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "fontSize": 30,
          "width": 200,
        }
      }
    >
      Enter account info here:
    </Text>
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "marginBottom": 5,
          "marginTop": 10,
        }
      }
    >
      User Name
    </Text>
    <TextInput
      allowFontScaling={true}
      name="user-name"
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 40,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "marginBottom": 5,
          "marginTop": 10,
        }
      }
    >
      Email
    </Text>
    <TextInput
      allowFontScaling={true}
      name="email"
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 40,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#f3f3f3",
          "marginBottom": 5,
          "marginTop": 10,
        }
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      name="password"
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 40,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 10,
          "shadowColor": "#f3f3f3",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 200,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f3f3f3",
                },
              ]
            }
          >
            sign up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
